ads/google/a4a/traffic-experiments.js

```diff
@@ -67,39 +73,45 @@ const INTERNALLY_SELECTED_ID = '2088462';
-    // randomlySelectUnsetPageExperiments won't add new IDs if
-    // maybeSetExperimentFromUrl has already set something for this
-    // experimentName.
-    randomlySelectUnsetPageExperiments(win, experimentInfo);
-    if (isExperimentOn(win, experimentName)) {
-      // Page is selected into the overall traffic experiment.
-      const selectedBranch = getPageExperimentBranch(win, experimentName);
-      addExperimentIdToElement(selectedBranch, element);
-      // Detect how page was selected into the overall experimentName.
-      if (isSetFromUrl) {
-        addExperimentIdToElement(EXTERNALLY_SELECTED_ID, element);
-      } else {
-        // Must be internally selected.
-        addExperimentIdToElement(INTERNALLY_SELECTED_ID, element);
-      }
-      // Detect whether page is on the "experiment" (i.e., use A4A rendering
-      // pathway) branch of the overall traffic experiment or it's on the
-      // "control" (i.e., use traditional, 3p iframe rendering pathway).
-      return selectedBranch == internalBranches.experiment ||
-          selectedBranch == externalBranches.experiment ||
-          selectedBranch == MANUAL_EXPERIMENT_ID;
+  if (!isGoogleAdsA4AValidEnvironment(win)) {
+    // Serving location doesn't qualify for A4A treatment
+    return false;
+  }
+
+  const isSetFromUrl = maybeSetExperimentFromUrl(win, element,
+      experimentName, externalBranches.control,
+      externalBranches.experiment, MANUAL_EXPERIMENT_ID);
+  const experimentInfo = {};
+  experimentInfo[experimentName] = internalBranches;
+  // Note: Because the same experimentName is being used everywhere here,
+  // randomlySelectUnsetPageExperiments won't add new IDs if
+  // maybeSetExperimentFromUrl has already set something for this
+  // experimentName.
+  randomlySelectUnsetPageExperiments(win, experimentInfo);
+  if (isExperimentOn(win, experimentName)) {
+    // Page is selected into the overall traffic experiment.
+    // In other words, if pre-launch serve A4A, else don't.
+    const selectedBranch = getPageExperimentBranch(win, experimentName);
+    addExperimentIdToElement(selectedBranch, element);
+    // Detect how page was selected into the overall experimentName.
+    if (isSetFromUrl) {
+      addExperimentIdToElement(EXTERNALLY_SELECTED_ID, element);
+    } else {
+      // Must be internally selected.
+      addExperimentIdToElement(INTERNALLY_SELECTED_ID, element);
     }
+    // Detect whether page is on the "experiment" (i.e., use A4A rendering
+    // pathway) branch of the overall traffic experiment or it's on the
+    // "control" (i.e., use traditional, 3p iframe rendering pathway).
+    const selected = selectedBranch == internalBranches.experiment ||
+                     selectedBranch == externalBranches.experiment ||
+                     selectedBranch == MANUAL_EXPERIMENT_ID;
+    return (selected == !hasLaunchedDoubleClick(element)); // TODO Not necessarily DoubleClick!
```

Comment:
As I think through it, I'm not sure that this sense-flip is the right way to handle things.  Post-launch, we'll have three "experiment" branches:

1. Default/no eid: Use FF.
1. Holdback: Use DF; send "is holdback" eid.
1. Traffic balance for Holdback: Use FF, but send "is traffic balance" eid.

(1) is handled by the `else` block, below.  (2) and (3) could be handled by the existing logic, so long as you use the correct eids in the corresponding states.  And that can be accomplished by choosing whether you attach the new, "post-launch" eids to the control or experiment branches in the `amp-ad-network-*-impl/0.1/*config.js` files.